src/librawspeed/decompressors/VC5Decompressor.cpp

```diff
@@ -0,0 +1,614 @@
+/*
+    RawSpeed - RAW file decoder.
+
+    Copyright (C) 2018 Stefan Löffler
+
+    This library is free software; you can redistribute it and/or
+    modify it under the terms of the GNU Lesser General Public
+    License as published by the Free Software Foundation; either
+    version 2 of the License, or (at your option) any later version.
+
+    This library is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA
+*/
+
+/*
+ * The code in this file is based on the Apache2/MIT dual-licensed code of the
```

Comment:
Define `based`?
Is there copy-pasted code?
I really would want to avoid having non-LGPL code.

Reply:
No, there is no directly copy-pasted code. Only the table17.inc file and the definitions needed by it (_rlv, RLVTABLE) are copied verbatim and marked as such.
However, the algorithm is logic is the same; as I stated above, I did not have access to the original VC5 specs, so the code is a reimplementation of the gopro code and I would therefore consider it a derivative work.
If it helps, I could remove the explicit mentioning of the licensing, but I think the original code should still be credited.

Reply:
> No, there is no directly copy-pasted code. 

Great. Then i would prefer something much more vague.

"This is a decompressor for VC-5 raw compression algo, as used in GoPro raws.
This implementation is similar to that one of the official reference implementation in https://github.com/gopro/gpr project,
and is producing bitwise-identical output as compared with the Adobe DNG Converter implementation."

Or something, don't know.
But actually talking about the license, i would say, does not give any positive benefit for us.